### ISSUE TITLE
Custom default options

### ISSUE THREAD

**Issue by JadedBlueEyes** [getzola/zola#634]
# Feature

## Current Problem

Many people have wants and needs that are different to anyone else's, and we can't expect the default's to be perfect for everyone. However, Zola currently isn't very accomodating for sites that don't fit the mould. A big part of this will be things like custom output formats, pagination, and betterout of the box features. However users should also be presented with a way to configure their own defaults. They may want to use a different template for every page in a specific collection, or set a default author for personal blogs. This should be simple and easy.

## Description

[Jekyll demonstrates what many people would want perfectly.](https://jekyllrb.com/docs/configuration/front-matter-defaults/) Having a 'defaults' key under config, where you can set what specific file's front matter should default to.

**Comment by Keats** [comment]
>  A big part of this will be things like custom output formats, pagination, and better out of the box features.

Custom output format is being thought of: https://zola.discourse.group/t/proposal-custom-output-formats/68/2

Can you detail what is missing from the current pagination? Rather than putting everything in the config file, configuration happens at the section level. I have never used Jekyll so I might be wrong but it seems it doesn't have the concept fo sections so you have to put it in the configuration instead?

> They may want to use a different template for every page in a specific collection

Zola doesn't have collections, do you mean sections? If so, that is already possible

> or set a default author for personal blogs.

Zola doesn't have the concept of authors but you can set anything you want in the `extra` bit of the config: https://github.com/getzola/zola/blob/master/docs/config.toml#L12-L13

**Comment by JadedBlueEyes** [comment]
> Can you detail what is missing from the current pagination?

Mainly generating multiple pages. https://github.com/getzola/zola/issues/624 describes one usage for this.

> If so, that is already possible

How? If I wanted to set the template for all blog articles to "post", and `extra.author` to "SHYC", how would I do that? It doesn't appear to be documented.

> doesn't have the concept fo sections so you have to put it in the configuration instead

Jekyll has collections, what I got mixed up with, that are folders in the root. You can set options for them, and specific pages and patterns, in the config file, but you must store data in seperate files. This means all config is in one place, and the use of YAML makes it quite easy, especialy with the high level of editor support.

**Comment by Keats** [comment]
> Mainly generating multiple pages. #624 describes one usage for this.

That issue doesn't seem to be about pagination though?

>  If I wanted to set the template for all blog articles to "post"

The `page_template` variable described in https://www.getzola.org/documentation/content/section/

>  and extra.author to "SHYC

For that case, it would be in the `config` extra (like in my link above) as  `config` is passed in every template rendering. 

> This means all config is in one place

That doesn't seem really ideal to me, especially when you have many subsections, some transparent. That specific change of putting section level config in `config.toml` is not going to happen

**Comment by JadedBlueEyes** [comment]
> That issue doesn't seem to be about pagination though?

Yes, it's not about pagination, or anything else mentioned in that sentance. That chunk of paragraph was mostly context setting though.

>The `page_template` variable 

Ahh, ok. 
To be honest, though, that seems a bit janky. Perhaps a `page_defaults` section, that allows for things like this, and extra options, to be set more eligantly?

> For that case, it would be in the `config` extra

Wouldn't that create really janky templates? 

```
{%- set author = page.extra.author | default(value=config.extra.author) -%}
```
or 
```
{%- if page.extra.author -%}
{{ page.extra.author }}
{%- elif config.extra.author -%}
{{ config.extra.author }}
{%- endif -%}
```
> That doesn't seem really ideal to me, especially when you have many subsections, some transparent. That specific change of putting section level config in config.toml is not going to happen

Fair enough, this is more a matter of opinions and preference, but a choice is nice.

**Comment by Artoria2e5** [comment]
In the same vain, is there a way to slap a `insert_anchor_links` onto every section as a default? These sort of settings become annoying to copy-paste when a site ends up with many sections.

**Comment by Keats** [comment]
`insert_anchor_links` could probably be a config option in config.toml

**Comment by ctron** [comment]
As I want to migrate from Worpress to Zola, keeping a path structure of `yyyy/mm/dd` I need to create a "transparent" section for every day. Super annoying. But even more annoying is that it seems I can't add `insert_anchor_links` on the top section, but need to place it in every section.

I think having this as a default configuration for the site makes total sense.
